test/apps/iodemo/io_demo.cc

```diff
@@ -473,27 +475,33 @@ class P2pDemoCommon : public UcxContext {
     public:
         Buffer(void *buffer, size_t size, BufferMemoryPool<Buffer> &pool,
                ucs_memory_type_t memory_type, UcxContext *map_context,
-               ucp_mem_h memh) :
+               ucp_mem_h memh, bool zcopy = false) :
             _capacity(size),
             _buffer(buffer),
             _size(0),
             _pool(pool),
             _memory_type(memory_type),
             _map_context(map_context),
-            _memh(memh)
+            _memh(memh),
+            _zcopy(zcopy)
         {
         }
 
         static Buffer *allocate(size_t size, BufferMemoryPool<Buffer> &pool,
                                 ucs_memory_type_t memory_type,
-                                UcxContext *map_context)
+                                UcxContext *map_context, bool zcopy)
         {
 #ifdef HAVE_CUDA
             cudaError_t cerr;
 #endif
             ucp_mem_h memh;
             void *buffer;
 
+            if (zcopy) {
+                return new Buffer(NULL, size, pool, memory_type, NULL, NULL,
+                                  true);
```

Comment:
buffer = NULL then size also has to be 0, maybe separate constructors with different parameters for different use cases?